otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -178,70 +163,88 @@ caml_runtime_events_create_cursor(const char_os* runtime_events_path, int pid,
     0,
     0
   );
-
-  if( cursor->metadata == NULL ) {
-    caml_stat_free(cursor);
-    caml_stat_free(runtime_events_loc);
-    return E_MAP_FAILURE;
+  if (cursor->metadata == NULL) {
+    ret = E_MAP_FAILURE;
+    goto failed2;
   }
 
   cursor->ring_file_size_bytes = GetFileSize(cursor->ring_file_handle, NULL);
-#else
-  ring_fd = open(runtime_events_loc, O_RDONLY, 0);
+  return E_SUCCESS;
 
-  if( ring_fd == -1 ) {
-    caml_stat_free(cursor);
-    caml_stat_free(runtime_events_loc);
-    return E_OPEN_FAILURE;
-  }
+ failed2:
+  CloseHandle(cursor->ring_handle);
+ failed1:
+  CloseHandle(cursor->ring_file_handle);
+  return ret;
+#else
```

Comment:
The stylistic distinction between the Windows and non-Windows sides of this function is quite jarring.